### ISSUE TITLE
Match error placement is off

### ISSUE THREAD

**Issue by aisapatino** [aisapatino/sjfnw#745]
![match-error](https://cloud.githubusercontent.com/assets/1279315/6967698/d7064652-d915-11e4-99c5-9200d520685c.png)

Amount error is above the employer field and no error shows for employer (there should be something saying it's required, since amount was entered). Most likely the template needs adjusting.

**Comment by tarhata** [comment]
SO, I'm not sure if this is the way to do this, but spent a few hours on it and decided to just show you where I landed. Fixing the placement errors took like a minute, but making sure that the `Employer` field displayed an error when a negative `Amount matched` was input took some thinking. 

In essence, I removed the `min_value` attribute from the `match_expected` widget because I couldn't figure out how to display an error for the `Employer` field when a ValidationError is raised for the `match_expected` field. Tried googling and reading the [related docs] (https://docs.djangoproject.com/en/1.5/ref/forms/validation/#raising-multiple-errors), but I still couldn't figure it out another way.

Help please!

**Comment by aisapatino** [comment]
Oh, crap, that might be my mistake. If there's an error on the amount field, the validation process must abort before it gets to the required field checks. I'd say that's fine, given that it's not an easy fix, it's not that important. I wanted to make sure the employer error was working, and I'm guessing it shows up if you enter a valid amount?

So basically, just the template fix is good. Sorry for all that trouble!

**Comment by tarhata** [comment]
Haha, that's ok. I felt like I didn't know what I was doing! :sweat: 

The employer error shows up when a positive amount is input in the `amount` field, but nothing is  input in the `employer` field. So if switching the placement is all we need to do, I'll just fix that part of the template and call it good.

**Comment by tarhata** [comment]
Just out of curiosity, is the fix I did the way you would do this if you wanted an `Employer` error to show at the same time as a negative amount error? If a ValidationError aborts checking the fields, then how would you do this?

**Comment by aisapatino** [comment]
Sorry, I didn't have a chance to look closely at the code before, but yes, that is what I would do. From the docs (the ones you linked), I'm not really sure why the error wasn't showing, since my guess about errors aborting the process was wrong:

>once those two methods are run for every field, the Form.clean() method, or its override, is executed, no matter if the previous methods have raised errors or not.

But I think moving it down the chain of `clean`s was a good approach.

**Comment by aisapatino** [comment]
Oh, I just realized - probably when the `amount` value is invalid, and the individual field's methods detect that, it isn't included in `cleaned_data`. Which would prevent [these checks](https://github.com/aisapatino/sjfnw/commit/8537ded75797c39de014f60f9e33f03c7063f828#diff-e8cdb7c27aa812c62bc1b63e9b5eaedbL190) from taking effect.

Found this in [those docs](https://docs.djangoproject.com/en/1.5/ref/forms/validation/)

>By the time the form’s clean() method is called, all the individual field clean methods will have been run (the previous two sections), so self.cleaned_data will be populated with any data that has survived so far. So you also need to remember to allow for the fact that the fields you are wanting to validate might not have survived the initial individual field checks.

Sorry again - I feel bad that I accidentally sent you on a bit of a goose chase because I didn't remember these details. You taught us both some things about form validation if that's any consolation :grimacing:
